pageserver/src/tenant.rs

```diff
@@ -7013,10 +7013,14 @@ mod tests {
                     lsn_range: Lsn(0x30)..Lsn(0x31),
                     is_delta: false
                 },
-                // The delta layer that is cut in the middle
+                // The delta layer covers the full range
                 PersistentLayerKey {
-                    key_range: get_key(3)..get_key(4),
-                    lsn_range: Lsn(0x30)..Lsn(0x41),
+                    key_range: {
+                        let mut key = Key::MAX;
+                        key.field6 -= 1;
```

Comment:
Without any context from earlier PRs this looks like this range would contradict with the comment above.

Reply:
It does cover the full range, we just make it artificially smaller to avoid it being mistaken for an L0 layer. See the comment near `hack_end_key` (present in this diff but you need to fully expand the hidden files). It's nothing new, as `hack_image_layer_range` does the same on main.

If it were in a constant, it would be explained in an unified place.

Reply:
https://github.com/neondatabase/neon/pull/8608/commits/d6831d4c0feef7fc02d9b8af642aeda1f591eb58